pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -221,6 +220,17 @@ func (r *ReconcileBareMetalHost) Reconcile(request reconcile.Request) (result re
 			return reconcile.Result{Requeue: true}, nil
 		}
 		reqLogger.Info("No status cache found")
+	} else {
+		//The status annotation is unneeded, as the status is present, and it will get outdated, so removing it
+		objStatus, err := r.getHostStatusFromAnnotation(host)
+		if err == nil && objStatus != nil {
```

Comment:
I think this should be

    if err != nil || objStatus != nil {

A non-nil `err` can only occur if the annotation is present but cannot be unmarshalled. In this case we still want to delete it.

Actually, even better would be to just do:

    if _, ok := annotations[metal3v1alpha1.StatusAnnotation]; ok {

since that's the thing we actually care about, which saves us from unmarshalling the data for no reason.